### PR TITLE
fix: add type safety to sourcemap construction

### PR DESCRIPTION
**Pull request by mellyeliu**
## Context

Adds type-safe key matching in addSourceMapData using explicit Babel typeguards (`t.isIdentifier`, `t.isStringLiteral`, `t.isNumericLiteral`). This improves type safety and prevents runtime errors  in edge cases like object spreads (disallowed now, but theoretical) or computed keys, as the AST shape can vary and .name/value may be undefined

## Testing
```
npm run test -w @stylexjs/babel-plugin
```

## Pre-flight checklist

- [x] I have read the contributing guidelines
      [Contribution Guidelines](https://github.com/facebook/stylex/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

**Review by necolas — APPROVED** [review]
Would be good to have a transform to catch potential regressions here

File: packages/babel-plugin/src/utils/add-sourcemap-data.js [hunk 1]
```diff
@@ -74,10 +74,12 @@ export function addSourceMapData(
       .get('arguments.0.properties')
```
Comment (nmn) [inline packages/babel-plugin/src/utils/add-sourcemap-data.js:74]: Using this will remove the need for the `$Flowignore`
```
.get('arguments')[0].get('properties')
```
